server/controllers/comment.js

```diff
@@ -0,0 +1,104 @@
+const { GeneralError, NotFound, Unauthorized } = require('../utils/errors');
+const { validationResult } = require('express-validator');
+
+const Profile = require('../models/profile');
+const Post = require('../models/post');
+const Comment = require('../models/Comment');
+
+exports.addComment = async (req, res, next) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty())
+        return res.status(400).json({
+            error: errors.array(),
+        });
+
+    const { postId } = req.params;
+    const { userId, text } = req.body;
+
+    try {
+        const post = await Post.findById(postId);
+        if (!post) throw new NotFound('No post found');
+
+        const profile = await Profile.findOne({ user: userId });
+        if (!profile) throw new NotFound('No profile found');
+
+        const comment = new Comment({
+            user: userId,
+            text,
+            name: `${profile.firstName} ${profile.lastName}`,
+            avatar: profile.imageUrl,
```

Comment:
comment is missing a date property?

Reply:
It is in the model as default to Date.now.